dietpi/func/dietpi-banner-wrap.awk

```diff
@@ -0,0 +1,96 @@
+function make_whitespace_offset(clean_line){
+    ## Generate leading spaces based on global INDENT_TYPE
+    if (INDENT_TYPE == "min"){
+        return sprintf("%*s", INDENT_MIN, "")
+    }
+    lead_spaces = ""
+    if ((INDENT_TYPE == "dash" || INDENT_TYPE == "dashcolon") && match(clean_line, green_dash)){
+        lead_spaces = sprintf("%*s", RSTART + RLENGTH - 1, "")
+    }
+    if ((INDENT_TYPE == "colon" || INDENT_TYPE == "dashcolon") && match(clean_line, green_colon)){
+        lead_spaces = sprintf("%*s", RSTART + RLENGTH - 1, "")
+    }
+    # Useable space check
+    # - "Let's Encrypt cert status" would be squashed otherwise
+    if ( (MAXCOL - length(lead_spaces)) < MIN_USEABLE_SPACE){
+        lead_spaces = sprintf("%*s", INDENT_MIN, "")
+    }
+    return lead_spaces
+}
+
+BEGIN {
+    green_dash = "^[[:space:]]+-[[:space:]]"  # green "bullets"
+    green_colon = "[[:space:]]:[[:space:]]"  # green colon in middle
+    color_regex = "[[:cntrl:]][[0-9;?]*[A-Za-z]"  # all color codes
+    ## Init from CLI, or use defaults here
+    if (MIN_USEABLE_SPACE == ""){MIN_USEABLE_SPACE = 8} ## min space on the right to keep
+    if (MAXCOL == ""){MAXCOL = 30}                      ## Wrap to column number
+    if (INDENT_MIN == ""){INDENT_MIN = 1}               ## min offset seen in dietpi banner
+    if (INDENT_TYPE == ""){INDENT_TYPE = "dashcolon"}   ## Indent modes:
+    ## min - use minimum
+    ## dash - green_dash only
+    ## colon - green_colon only
+    ## dashcolon - match dash first then colon (colon overrides)
+}
+
+{
+    ## ASCII ART: Skip or Hide
+    if (match($0, /^[^a-zA-Z0-9─]+$/)){
+        if (MAXCOL > (RSTART+RLENGTH)){print $0;}
+        next;
+    }
+    ## Green Lines: Truncate to MAXCOL
+    if (match($0, /────/)){
+        new_grn_line = sprintf("%*s", MAXCOL - 2, "") ## new line: of spaces
```

Comment:
Maybe we should cap it at 54, which is the current right side column if word-wrapping is not enabled? So the lines do not change on larger terminals, but only if smaller than 54 columns.

`MAXCOL` itself of should needs to remain `tput cols`, to hide the ASCII art hostname and wrap other larger output lines correctly. Just for these green separator lines the cap would make sense.